source/Octopus.Client.E2ETests/AssemblyExtensions.cs

```diff
@@ -1,19 +1,22 @@
 ﻿using System;
 using System.IO;
 using System.Reflection;
-using Octopus.Client.Model;
 
 namespace Octopus.Client.E2ETests
 {
     internal static class AssemblyExtensions
     {
         public static string FullLocalPath(this Assembly assembly)
         {
-            var codeBase = assembly.CodeBase;
+#if NETFRAMEWORK
+            var codeBase = assembly.CodeBase ?? throw new NotSupportedException($"Cannot get codebase for assembly {assembly}");
+#else
+            var codeBase = assembly.Location;
+#endif
```

Comment:
Could we just use `Assembly.Location` for both frameworks?

Reply:
We *most likely* can.

[Microsoft's documentation](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=net-7.0) contains this:

> **Remarks** 
In .NET 5 and later versions, for bundled assemblies, the value returned is an empty string.
>
> .NET Framework only: If the loaded file was [shadow-copied](https://learn.microsoft.com/en-us/dotnet/framework/app-domains/shadow-copy-assemblies), the location is that of the file after being shadow-copied. To get the location before the file has been shadow-copied, use the [CodeBase](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.codebase?view=net-7.0) property.

Out of caution, I chose to leave the code untouched for NETFRAMEWORK.